specifications/desktop-extensions-font-hook.org

```diff
@@ -0,0 +1,74 @@
+#+TITLE: Desktop Extensions Font Hook
+#+AUTHOR: Sergio Schvezov <sergio.schvezov@canonical.com>
+#+DATE: [2020-09-22]
+
+* Scope
+
+Setup the general font cache in the desktop extensions using an
+extension provided =configure= hook to cache the system fonts and let
+the /launcher/ only process /user fonts/ (e.g.; =.fonts=) to decouple font
+caching issues found on many systems and offload first run times to
+the system (e.g.; snapd install and refreshes).
+
+The cross distro story shall benefit from decoupling font processing
+from the system as there will be no more font cache misses due to
+cache ABI changes between the cache generation logic found within the
+snap and that found on the system.
+
+* General mechanism
+
+The =fonts.conf= file shall be modified to achieve fallback
+mechanisms.
+
+When the =configure= hook runs it must run =fc-cache= to update the fonts
+cache with from fonts that are generally available on the system using
+a privately generated =fonts.conf=. This cache must reside in a location
+that is generally available to the snap when run by a desktop user, to
+achieve such result, the cache shall reside in =$SNAP_DATA=.
+
+When the desktop appliaction launches, the launcher generated by the
+extension shall contain a reference to this cache and only process
+fonts that are installed on the user's private font instance, =~/.fonts=.
```

Comment:
Is .fonts accessible as a result of the desktop interface?

Reply:
Desktop interface is required to reach the system fonts